doc/sailfish-webview-webenginesettings.h

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL /2.0/. */
+
+#include <QObject>
+
+// This file is used only for documentation purposes
+
+// These items are all part of QMozEngineSettings, but inherited by
+// WebEngineSettings, so we take some notational liberty and present them
+// as part of the the subclass.
+
+namespace SailfishOS {
+
+class WebEngineSettings : public QMozEngineSettings {
+    Q_OBJECT
+    Q_PROPERTY(bool autoLoadImages READ autoLoadImages WRITE setAutoLoadImages NOTIFY autoLoadImagesChanged FINAL)
+    Q_PROPERTY(bool javascriptEnabled READ javascriptEnabled WRITE setJavascriptEnabled NOTIFY javascriptEnabledChanged FINAL)
+    Q_PROPERTY(bool popupEnabled READ popupEnabled WRITE setPopupEnabled NOTIFY popupEnabledChanged)
+    Q_PROPERTY(CookieBehavior cookieBehavior READ cookieBehavior WRITE setCookieBehavior NOTIFY cookieBehaviorChanged)
+    Q_PROPERTY(bool useDownloadDir READ useDownloadDir WRITE setUseDownloadDir NOTIFY useDownloadDirChanged)
+    Q_PROPERTY(QString downloadDir READ downloadDir WRITE setDownloadDir NOTIFY downloadDirChanged)
+    Q_PROPERTY(qreal pixelRatio READ pixelRatio WRITE setPixelRatio NOTIFY pixelRatioChanged)
+    Q_PROPERTY(bool initialized READ isInitialized NOTIFY initialized)
+
+public:
+    // C++ API
+    static void initialize();
+    static WebEngineSettings *instance();
+
+    explicit WebEngineSettings(QObject *parent = 0);
+    virtual ~WebEngineSettings();
+
+    // See developer.mozilla.org/en-US/docs/Mozilla/Firefox/Privacy/Storage_access_policy
+    // And git.sailfishos.org/mer-core/gecko-dev/blob/master/netwerk/cookie/nsCookieService.cpp
+    enum CookieBehavior {
+        AcceptAll = 0,
+        BlockThirdParty = 1,
+        BlockAll = 2,
+        Deprecated = 3
+    };
+    Q_ENUM(CookieBehavior)
```

Comment:
I'm now a bit puzzled about this. For QML side of the WebEngineSettings I think it's documented exactly right. 

```
main.cpp:15:72: error: conversion from ‘QMozEngineSettings::CookieBehavior’ to non-scalar type ‘SailfishOS::WebEngineSettings’ requested
     SailfishOS::WebEngineSettings foo = SailfishOS::WebEngineSettings::BlockAll;
                                                                        ^~~~~~~~
```
Like expected with this snippet
```
SailfishOS::WebEngineSettings foo = SailfishOS::WebEngineSettings::BlockAll;
SailfishOS::WebEngineSettings::instance()->setCookieBehavior(SailfishOS::WebEngineSettings::BlockAll);
```

Same work if int used instead
```
int foo = SailfishOS::WebEngineSettings::BlockAll;
SailfishOS::WebEngineSettings::instance()->setCookieBehavior(SailfishOS::WebEngineSettings::BlockAll);
```

I think we should document this enum as as an of QMozEngineSettings (fake class for it).

Reply:
I may be misunderstanding here so please bear with me, but I'm finding this code is working okay:
```
    SailfishOS::WebEngineSettings::CookieBehavior foo = SailfishOS::WebEngineSettings::BlockAll;
    SailfishOS::WebEngineSettings::instance()->setCookieBehavior(SailfishOS::WebEngineSettings::BlockAll);
```

Reply:
It was my quick copy-paste... Sorry for the noise.